state/endpoint_bindings.go

```diff
@@ -294,14 +323,14 @@ func validateEndpointBindingsForMachines(a *Application, oldBindings, newBinding
 		// it will not have a provider address in the default
 		// space so the machine-count check below would
 		// otherwise fail.
-		if spName == network.DefaultSpaceName {
+		if spID == network.DefaultSpaceId {
 			continue
 		}
 
 		// Ensure that all currently deployed machines have an address
 		// in the requested space for this binding
-		if machineCountInSpace[spName] != len(deployedMachines) {
-			return errors.Errorf("binding endpoint %q to space %q is not feasible: one or more deployed machines lack an address in this space", epName, spName)
+		if machineCountInSpace[spID] != len(deployedMachines) {
+			return errors.Errorf("binding endpoint %q to space %q is not feasible: one or more deployed machines lack an address in this space", epName, spID)
```

Comment:
This error should include the space name as it gets returned back to the API caller.

Reply:
updated